src/dragonball/src/device_manager/legacy.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright (C) 2022 Alibaba Cloud. All rights reserved.
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Portions Copyright 2017 The Chromium OS Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the THIRD-PARTY file.
+
+//! Device Manager for Legacy Devices.
+
+use std::io;
+use std::sync::{Arc, Mutex};
+
+use dbs_device::device_manager::Error as IoManagerError;
+use dbs_legacy_devices::SerialDevice;
+use vmm_sys_util::eventfd::EventFd;
+
+/// Errors generated by legacy device manager.
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    /// Cannot add legacy device to Bus.
+    #[error("failure while managing legacy device")]
```

Comment:
This is rather a generic description. How about something like:

```suggestion
    #[error("Bus error while managing legacy device")]
```

Reply:
Changed